app/models/concerns/smooch_resend.rb

```diff
@@ -219,16 +219,17 @@ def template_locale_options(team_slug = nil)
     end
 
     def format_template_message(template_name, placeholders, file_url, fallback, language, file_type = 'image', preview_url = true)
+      provider = RequestStore.store[:smooch_bot_provider]
       namespace = self.config['smooch_template_namespace']
       template = self.config["smooch_template_name_for_#{template_name}"] || template_name
-      return '' if namespace.blank? || template.blank?
+      return '' if ['TURN', 'CAPI'].include?(provider) && (namespace.blank? || template.blank?)
```

Comment:
Templates are only for WhatsApp. So this verification is to be sure that execution can continue and the message can be sent for non-WhatsApp tiplines even if there is no template namespace and template name defined.